From de351477979e962f65eec8836f9315bb31697f8c Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Mon, 4 Jan 2021 03:38:14 -0500
Subject: [PATCH] HACK: always show logo

It seems showing the logo is not a priority anymore to the linux kernel,
and I can't figure out where in the whole stack it gets set to
FBCON_LOGO_DONTSHOW. I'm thinking it's probably via a do_fbcon_takeover
with (0) (show_logo) that it breaks.

Oh well, it's mainly for my own enjoyment, so for now I'll live with
that. I should probably trace out from where it comes from later on.
---
 drivers/video/fbdev/core/fbcon.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/video/fbdev/core/fbcon.c b/drivers/video/fbdev/core/fbcon.c
index cef437817b0dc..0d21da0fd4e33 100644
--- a/drivers/video/fbdev/core/fbcon.c
+++ b/drivers/video/fbdev/core/fbcon.c
@@ -1044,8 +1044,7 @@ static void fbcon_init(struct vc_data *vc, int init)
 	info = registered_fb[con2fb_map[vc->vc_num]];
 	cap = info->flags;
 
-	if (logo_shown < 0 && console_loglevel <= CONSOLE_LOGLEVEL_QUIET)
-		logo_shown = FBCON_LOGO_DONTSHOW;
+	logo_shown = FBCON_LOGO_CANSHOW;
 
 	if (vc != svc || logo_shown == FBCON_LOGO_DONTSHOW ||
 	    (info->fix.type == FB_TYPE_TEXT))
-- 
2.29.2

